src/document.cpp

```diff
@@ -355,6 +355,20 @@ rvoe<NoReturnValue> PdfDocument::add_page(std::string resource_dict,
     return NoReturnValue{};
 }
 
+rvoe<NoReturnValue> add_page_label(PageId page_id, std::string label) {
+    page_labels.append({page_id, CAPY_PAGE_LABEL_NUMBER_STYLE_NONE, label, 0});
+    return NoReturnValue{};
```

Comment:
This could be implemented by calling the other method with specific parameters. Then there would be only once entry point where the array is manipulated.

But from the public API perspective there should probably be only one C function exposed. One of the design goals of the  C API has been to minimize the number of exposed functions. A function that could be implemented by calling a different function with specific (known-beforehand) arguments is not really that.

Reply:
Yes, this is where it gets complicated. I was going to expose the `labeling scheme` in the API while having an optional label string attribute in the new page function for the other workflow.